### PR TITLE
chore(iatlas): migrate iAtlas API project to an Nx project (ARCH-339)

### PR DESCRIPTION
**Pull request by tschaffter**
Closes [ARCH-339](https://sagebionetworks.jira.com/browse/ARCH-339)

## Changelog

- Migrate `apps/iatlas/api` to an Nx project.
- Create a Poetry project to manage the dependencies.
- Rewrite the Dockerfile from a standard Python image.
- Add a Docker Compose file for the command `nx serve-detach`.

## Preview

image

[ARCH-339]: https://sagebionetworks.jira.com/browse/ARCH-339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/iatlas/api/poetry.lock [hunk 1]
```diff
@@ -0,0 +1,805 @@
+# This file is automatically @generated by Poetry 1.8.3 and should not be changed by hand.
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply apps/iatlas/api/poetry.lock:1]: This issue is beyond the scope of this PR, but we can address it separately.

[the remark this answers is not in the document]
Comment (tschaffter) [reply apps/iatlas/api/poetry.lock:1]: This issue is beyond the scope of this PR, but we can address it separately.

[the remark this answers is not in the document]
Comment (tschaffter) [reply apps/iatlas/api/poetry.lock:1]: This issue is beyond the scope of this PR, but we can address it separately.

[the remark this answers is not in the document]
Comment (tschaffter) [reply apps/iatlas/api/poetry.lock:1]: This issue is beyond the scope of this PR, but we can address it separately.

[the remark this answers is not in the document]
Comment (tschaffter) [reply apps/iatlas/api/poetry.lock:1]: This issue is beyond the scope of this PR, but we can address it separately.

File: apps/iatlas/api/poetry.lock [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply apps/iatlas/api/poetry.lock:1]: This issue is beyond the scope of this PR, but we can address it separately.

**Review by andrewelamb — APPROVED** [review]

**Comment by andrewelamb** [comment]
@tschaffter This looks good to me, excited to try it out!

**Pull request merged**